src/poller/poller-core/index.js

```diff
@@ -367,67 +406,110 @@ async function parseAndEnrichPayload(payload) {
   const clusters = await configValidator.parseAndAssertValidConfig(payload);
   const clustersFound = [];
 
-  for (let idx = 0; idx < clusters.length; idx++) {
+  for (let clusterIdx = 0; clusterIdx < clusters.length; clusterIdx++) {
+    const customMetrics =
```

Comment:
Is there a reason why this is object is assigned here, but only used ~50 lines down?

Reply:
The `metrics` structure is used to hold any custom metrics from the config as well as the default/enriched metrics, so we want this before any of the modifications have been made. If there were no custom metrics then this will be null farther down when tested.